internal/configs/annotations.go

```diff
@@ -295,6 +298,13 @@ func parseAnnotations(ingEx *IngressEx, baseCfgParams *ConfigParams, isPlus bool
 		}
 	}
 
+	if basicSecret, exists := ingEx.Ingress.Annotations[BasicAuthSecretAnnotation]; exists {
```

Comment:
the annotations names look good.

Could you add validation of their values here internal/k8s/validation.go ?

nginx.org/basic-auth-realm, of type string, required. validation: (1) must have all '"' (double quotes) escaped and must not end with an unescaped '\' (backslash) (2) must not have any variables.  also requires nginx.org/basic-auth-secret to be set.

nginx.org/basic-auth-secret, of type string, required. reference to the secret within the same namespace. validation: valid secret name. also requires nginx.org/basic-auth-realm to be set.

Reply:
an update based on the conversation about the realm field of the policy:

nginx.org/basic-auth-realm can be optional and can have value "".  However, it is still cannot be used without nginx.org/basic-auth-secret.